newrelic/data_source_newrelic_notifications_destination.go

```diff
@@ -83,7 +93,7 @@ func dataSourceNewRelicNotificationDestinationRead(ctx context.Context, d *schem
 
 	if len(destinationResponse.Entities) == 0 {
 		d.SetId("")
-		return diag.FromErr(fmt.Errorf("the id '%s' does not match any New Relic notification destination", id))
+		return diag.FromErr(fmt.Errorf("the id or name you provided does not match any New Relic notification destination"))
```

Comment:
Another suggestion - do you think it would be better if we can conditionally vary the error message printed based on the ID or Name? Something similar to 

```go
		if !idOk || idValue == "" {
			return diag.FromErr(fmt.Errorf("the id provided does not match any New Relic notification destination"))
		}
		if !nameOk || nameValue == "" {
			return diag.FromErr(fmt.Errorf("the name provided does not match any New Relic notification destination"))
		}
```

(just a sample, doesn't necessarily have to be the right solution)
Also, discarding **you** from the message would be ideal.

Reply:
done

Reply:
A small question again, @lzaga-newrelic - have you been able to test this after you made the change needed to address this comment? I just want to be sure the condition we discussed here (and you added to the code) is actually right and we're not missing anything

Reply:
just a small fix needed here - will push it right away

Reply:
@lzaga-newrelic maybe the new condition you've added needs to be the other way round 🤔 

`if !idOk || (idValue == "" && nameValue != "")`

and similarly for the name

(unless I'm missing something - please let me know)

Reply:
nope, because there is an id for example but incorrect so 
`if !idOk || (idValue != "" && nameValue == "")` will be the right way to go

Reply:
Oh yeah, got you 👍